Sources/BraintreePayPal/BTPayPalApprovalURLParser.swift

```diff
@@ -0,0 +1,44 @@
+import Foundation
+
+#if canImport(BraintreeCore)
+import BraintreeCore
+#endif
+
+/// The type of PayPal authentication flow to occur
+enum PayPalRedirectType {
+    
+    /// The in-app browser (ASWebAuthenticationSession) web checkout flow
+    case webBrowser(url: URL)
+    
+    /// The universal link flow, switching out of the merchant app into the native PayPal app
+    case payPalApp(url: URL)
+}
+
+/// Parses response body from `/v1/paypal_hermes/*` POST requests to determine the `PayPalRedirectType`
+struct BTPayPalApprovalURLParser {
+    
+    var redirectType: PayPalRedirectType
+    
+    var pairingID: String? {
+        switch redirectType {
+        case .webBrowser(let url), .payPalApp(let url):
+            let url = URLComponents(url: url, resolvingAgainstBaseURL: true)
+            if let token = url?.queryItems?.first(where: { $0.name == "token" || $0.name == "ba_token" })?.value,
```

Comment:
General question, is there ever an instance where we'd get both of these params back for the app switch flow but only actually care about 1 of them? Asking because if so we may always want to attempt to parse the `ba_token` first then if not there move on to the `token`.

Reply:
Yes! We should prioritize the `ba_token` over the `token` if both are available. Since this change also needs to be made on `main` (+ tests added for the priority), I was thinking it could be a separate PR targeting `main` first, and then handling the merge conflicts when updating this branch. Thoughts?

Reply:
Yeah I am fine if we want to target that on main instead of here. The merge conflict may be a bit weird since it'll likely be in the section we removed in this PR, but the tests would help us remember.